### PR TITLE
cdc: add cyclic replication doc

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed, added or deleted? (Required)

Add document about TiCDC cyclic replication feature.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: N/A
- Other reference link(s): N/A

### CODE REVIEW

**Comment by overvenus** [comment]
PTAL, thanks!

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -335,3 +335,110 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+        --addr=0.0.0.0:8301 \
+        --advertise-addr=127.0.0.1:8301
+    ```
+
+2. 在 TiDB 集群 A，B 和 C 上创建环形同步需要使用的标记数据表 (`mark table`)。
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    # 在 TiDB 集群 A 上创建表数据表。
```
Comment (amyangfei) [inline ticdc/manage-ticdc.md:393]:
```suggestion
    # 在 TiDB 集群 A 上创建标记数据表。
```
ditto for following two comments

File: ticdc/manage-ticdc.md [hunk 2]
```diff
@@ -335,3 +335,110 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+        --cyclic-filter-replica-ids 2 \
+        --cyclic-sync-ddl true
+
+    # 在 TiDB 集群 B 上创建环形同步任务。
+    cdc cli changefeed create \
+        --sink-uri="mysql://root@${TiDB_C_HOST}/" \
+        --pd="http://${PD_B_HOST}:${PD_B_PORT}" \
+        --cyclic-replica-id 2 \
+        --cyclic-filter-replica-ids 3 \
+        --cyclic-sync-ddl false
```
Comment (amyangfei) [inline ticdc/manage-ticdc.md:425]: 这里和示例图不一致，B 也是需要开启 sync-ddl 的？

File: ticdc/manage-ticdc.md [hunk 3]
```diff
@@ -335,3 +335,110 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+        --sink-uri="mysql://root@${TiDB_A_HOST}/" \
+        --pd="http://${PD_C_HOST}:${PD_C_PORT}" \
+        --cyclic-replica-id 3 \
+        --cyclic-filter-replica-ids 1 \
+        --cyclic-sync-ddl false
+    ```
+
+### 环形同步使用限制
+
+1. 环形复制功能使用到的表急表必须在创建环形同步任务前使用 `cdc cli changefeed cyclic create-marktables` 创建。
```
Comment (amyangfei) [inline ticdc/manage-ticdc.md:438]:
```suggestion
1. 环形复制功能使用到的标记表必须在创建环形同步任务前使用 `cdc cli changefeed cyclic create-marktables` 创建。
```

File: ticdc/manage-ticdc.md [hunk 4]
```diff
@@ -335,3 +335,110 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+2. 开启环形复制的数据表只包含 [a-zA-z0-9_] 字符。
+3. 开启环形复制的数据表表必须在创建环形同步任务前创建完毕。
+4. 开启环形复制后，不能创建一个会被环形同步任务同步的表。
+5. 如果想在线 DDL，需要确保：
+    1. 多个集群中有且仅有一个集群的 CDC(s) 开启了 cyclic-sync-ddl，例如示例中的 A 集群 CDC
```
Comment (amyangfei) [inline ticdc/manage-ticdc.md:443]: 在线 DDL 是需要保证“构成一个单向同步链表，不能成环” ？

File: ticdc/manage-ticdc.md [hunk 5]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
 ```
 election: not leader
 ```
+
+## 环形同步
+
+> **警告：**
+>
+> 环形同步功能还在实验中，没有经过完备的测试。暂时请避免在生产环境中使用该功能。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:343]:
```suggestion
> 目前环形同步属于实验特性，尚未经过完备的测试，不建议在生产环境中使用该功能。
```

File: ticdc/manage-ticdc.md [hunk 6]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
+环形同步功能可以在多个独立的 TiDB 集群间同步数据。比如有三个 TiDB 集群 A，B 和 C，它们都有一个数据表 `test.user_data`，并且各自对它有写入。环形同步功能可以将 A，B 和 C 对 `test.user_data` 的写入同步其它集群上，使三个集群上的 `test.user_data` 达到最终一致。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:345]:
```suggestion
环形同步功能支持在多个独立的 TiDB 集群间同步数据。比如有三个 TiDB 集群 A、B 和 C，它们都有一个数据表 `test.user_data`，并且各自对它有数据写入。环形同步功能可以将 A、B 和 C 对 `test.user_data` 的写入同步其它集群上，使三个集群上的 `test.user_data` 达到最终一致。
```

File: ticdc/manage-ticdc.md [hunk 7]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
+### 环形同步使用示例
+
+在三个集群 A, B 和 C 上开启环形复制，其中 A 到 B 的同步使用两个 TiCDC。A 作为
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:349]: `A 作为` 后面话没说完？
```suggestion
在三个集群 A、B 和 C 上开启环形复制，其中 A 到 B 的同步使用两个 TiCDC。
```

File: ticdc/manage-ticdc.md [hunk 8]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
+![TiCDC cyclic replication](/media/cdc-cyclic-replication.png)
+
+环形复制的使用需要设置同步任务的创建参数：
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:353]:
```suggestion
使用环形同步功能时，需要设置同步任务的创建参数：
```

File: ticdc/manage-ticdc.md [hunk 9]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
++ `--cyclic-replica-id`，为上游集群的写入指定来源 ID，需要确保每个集群的 ID 唯一性。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:355]:
```suggestion
+ `--cyclic-replica-id`：用于指定为上游集群的写入指定来源 ID，需要确保每个集群 ID 的唯一性。
```

File: ticdc/manage-ticdc.md [hunk 10]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
++ `--cyclic-filter-replica-ids`，指定需要过滤的写入来源 ID，通常为下游集群的 ID。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:356]:
```suggestion
+ `--cyclic-filter-replica-ids`：用于指定需要过滤的写入来源 ID，通常为下游集群的 ID。
```

File: ticdc/manage-ticdc.md [hunk 11]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
++ `--cyclic-sync-ddl`，是否同步 DDL 到下游，只能在一个集群的 CDC 上开启同步 DDL。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:357]:
```suggestion
+ `--cyclic-sync-ddl`：用于指定是否同步 DDL 到下游，只能在一个集群的 CDC 上开启 DDL 同步。
```

File: ticdc/manage-ticdc.md [hunk 12]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+
+环形同步任务创建步骤如下：
+
+1. 在 TiDB 集群 A，B 和 C 上[启动 TiCDC 组件](#TiCDC-部署)。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:361]:
```suggestion
1. 在 TiDB 集群 A，B 和 C 上[启动 TiCDC 组件](#ticdc-部署)。
```

File: ticdc/manage-ticdc.md [hunk 13]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+        --cyclic-replica-id 3 \
+        --cyclic-filter-replica-ids 1 \
+        --cyclic-sync-ddl false
+    ```
+
+### 环形同步使用限制
+
+1. 环形复制功能使用到的标记表必须在创建环形同步任务前使用 `cdc cli changefeed cyclic create-marktables` 创建。
+2. 开启环形复制的数据表只包含 [a-zA-z0-9_] 字符。
+3. 开启环形复制的数据表表必须在创建环形同步任务前创建完毕。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:443]: How about making it easier to read?
```suggestion
3. 在创建环形同步任务前，开启环形复制的数据表必须已创建完毕。
```

File: ticdc/manage-ticdc.md [hunk 14]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+        --sink-uri="mysql://root@${TiDB_A_HOST}/" \
+        --pd="http://${PD_C_HOST}:${PD_C_PORT}" \
+        --cyclic-replica-id 3 \
+        --cyclic-filter-replica-ids 1 \
+        --cyclic-sync-ddl false
+    ```
+
+### 环形同步使用限制
+
+1. 环形复制功能使用到的标记表必须在创建环形同步任务前使用 `cdc cli changefeed cyclic create-marktables` 创建。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:441]:
```suggestion
1. 在创建环形同步任务前，必须使用 `cdc cli changefeed cyclic create-marktables` 创建环形复制功能使用到的标记表。
```

File: ticdc/manage-ticdc.md [hunk 15]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+2. 开启环形复制的数据表只包含 [a-zA-z0-9_] 字符。
+3. 开启环形复制的数据表表必须在创建环形同步任务前创建完毕。
+4. 开启环形复制后，不能创建一个会被环形同步任务同步的表。
+5. 如果想在线 DDL，需要确保：
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:445]:
```suggestion
5. 如果想在线 DDL，需要确保以下两点：
```

File: ticdc/manage-ticdc.md [hunk 16]
```diff
@@ -335,3 +335,113 @@ curl -X POST http://127.0.0.1:8301/capture/owner/resign
...
+    1. 多个集群的 CDC 构成一个单向 DDL 同步链，不能成环，例如示例中只有 C 集群的 CDC 关闭 sync-ddl。
```
Comment (lilin90) [inline ticdc/manage-ticdc.md:446]:
```suggestion
    1. 多个集群的 CDC 构成一个单向 DDL 同步链，不能成环，例如示例中只有 C 集群的 CDC 关闭了 sync-ddl。
```

**Pull request merged**